metaflow/plugins/argo/argo_cli.py

```diff
@@ -0,0 +1,185 @@
+import click
+import json
+
+from metaflow import current, decorators, parameters, JSONType
+from metaflow.datastore.datastore import TransformableObject
+from metaflow.metaflow_config import from_conf
+from metaflow.package import MetaflowPackage
+from metaflow.plugins import BatchDecorator
+from .argo_workflow import ArgoWorkflow
+from .argo_exception import ArgoException
+
+
+@click.group()
+def cli():
+    pass
+
+
+@cli.group(help="Commands related to Argo Workflows.")
+@click.pass_obj
+def argo(obj):
+    obj.check(obj.graph, obj.flow, obj.environment, pylint=obj.pylint)
+
+
+@argo.command(help="Deploy a new version of this workflow to "
+                   "Argo Workflow Templates.")
+@click.option("--image",
+              default=None,
+              help="Docker image requirement in name:version format.")
+@click.option("--env",
+              default=None,
+              help="Environment variables to be set for the workflow.")
+@click.option("--env-from",
+              default=None,
+              help="Environment variables to be set for the workflow.")
+@click.option("--token",
+              default=None,
+              help="Authentication token to call Argo Server.")
+@click.option('--namespace',
+              default=None,
+              help="Deploy into the specified kubernetes namespace.")
```

Comment:
This may get confusing: `step-functions create` also supports `--namespace`, but in that case this is Metaflow namespace

Reply:
yes, the semantics differs a little.  But the name "namespace" is so common in k8s and using k8s namespaces is exactly how we separate workloads of different users in k8s cluster ...  hard to come up with a better name.
Integration with metaflow's namespaces isn't completed though.

Reply:
Yeah it is unfortunate, maybe something like `--k8s-namespace`? It is cumbersome but unambiguous

Reply:
done